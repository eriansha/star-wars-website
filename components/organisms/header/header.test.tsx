import { cleanup, fireEvent, render } from '@testing-library/react'

//import userEvent from '@testing-library/user-event'
import Header from './header'

afterEach(cleanup)

describe('Footer', () => {
  it('renders without crashing', () => {
    render(<Header />)
  })

  it('has \'sticky\' and \'top-0\' in header tag', () => {
    const { getByTestId } = render(<Header />)
    expect(getByTestId('main-header').classList.contains('sticky')).toBe(true)
    expect(getByTestId('main-header').classList.contains('top-0')).toBe(true)
  })

  it('does not have \'-translate-x-full\' in sidebar classname when initialize header', () => {
    const { getByTestId } = render(<Header />)
    expect(getByTestId('sidebar').classList.contains('-translate-x-full')).toBe(
      true
    )
  })

  it('has \'-translate-x-full\' in sidebar classname when user click sidebar button', () => {
    const { getByTestId } = render(<Header />)
    const sidebar = getByTestId('sidebar')

    fireEvent.click(getByTestId('sidebar-btn'))
    expect(sidebar.classList.contains('-translate-x-full')).toBe(false)
  })

  // TODO: fix unit test
  // it('contains image', () => {
  //   const alt='starwars banner'
  //   const src = '/img/brand.png'

  //   const tree = render(<Header />)
  //   const img = tree.getByAltText(alt)
  //   expect(img.getAttribute('src')).toBe(src);
  // })

  it('matches snapshot', () => {
    const tree = render(<Header />)
    expect(tree.container).toMatchSnapshot()
  })
})
