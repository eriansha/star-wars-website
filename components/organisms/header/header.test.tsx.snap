// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
<div>
  <header
    class="bg-gray-dark border-b-2 border-gray flex justify-start sticky top-0 z-10"
    data-testid="main-header"
  >
    <div
      class=""
    >
      <a
        class="block"
        href="/"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 150px; height: 75px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <img
            alt="starwars banner"
            data-nimg="fixed"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </a>
    </div>
    <div
      class="hidden items-center justify-start md:flex"
    >
      <nav>
        <a
          class="px-4 py-2"
          href="/films"
        >
          Movie
        </a>
      </nav>
      <nav>
        <a
          class="px-4 py-2"
          href="/people"
        >
          People
        </a>
      </nav>
      <nav>
        <a
          class="px-4 py-2"
          href="/planet"
        >
          Planet
        </a>
      </nav>
      <nav>
        <a
          class="px-4 py-2"
          href="/species"
        >
          Species
        </a>
      </nav>
      <nav>
        <a
          class="px-4 py-2"
          href="/starship"
        >
          Starship
        </a>
      </nav>
    </div>
    <button
      class="ml-auto p-4 md:hidden focus:bg-gray-700 focus:outline-none"
      data-testid="sidebar-btn"
    >
      <svg
        class="h-5 w-5"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
        />
      </svg>
    </button>
  </header>
  <div
    class="absolute bg-gray-dark duration-200 ease-in-out inset-y-0 left-0 px-2 space-y-6 text-blue-100 transform transition w-64 z-20 md:hidden md:relative md:translate-x-0 -translate-x-full"
    data-testid="sidebar"
  >
    <a
      class="flex items-center px-2 space-x-2"
      href="/"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 200px; height: 100px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <img
          alt="starwars banner"
          data-nimg="fixed"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </a>
    <nav>
      <a
        class="block px-4 py-2"
        href="/films"
      >
        Movie
      </a>
    </nav>
    <nav>
      <a
        class="block px-4 py-2"
        href="/people"
      >
        People
      </a>
    </nav>
    <nav>
      <a
        class="block px-4 py-2"
        href="/planet"
      >
        Planet
      </a>
    </nav>
    <nav>
      <a
        class="block px-4 py-2"
        href="/species"
      >
        Species
      </a>
    </nav>
    <nav>
      <a
        class="block px-4 py-2"
        href="/starship"
      >
        Starship
      </a>
    </nav>
  </div>
</div>
`;
